<?xml version="1.0"?>
<launch>
    <!-- ============================ SETTINGS ============================= -->

    <!-- Node frequency: the frequency [Hz] which rate.sleep() function will
    attempt to maintain in the main_loop. -->
    <arg 
    name="node_frequency" 
    default="100"/>

    <!-- Start position (chest): move chest to starting position
    [meters]: [0.0 - 0.44]. If set to -1, no starting position is set and the
    chest stays at current position. -->
    <arg 
    name="start_position" 
    value="0.22"/>

    <!-- Arm center of workspace Z: the center of arm's workspace in GCS Z
    coordinate. This value is used for Proximity activation boundaries
    calculation. -->
    <arg 
    name="arm_center_of_workspace_z" 
    value="0.2"/>

    <!-- Activation boundary Z: Proximity activation boundaries (in GCS) are
    calculate in the following way: 
        'min': arm_center_of_workspace_z - activation_boundary_z
        'max': arm_center_of_workspace_z + activation_boundary_z -->
    <arg 
    name="activation_boundary_z" 
    value="0.3"/>


    <!-- If any of dependency or optional subfiles should not be launched (or
    are launched separately) set corresponding argument to "false". -->
    <arg 
    name="launch_chest" 
    default="true"/>
    <arg 
    name="launch_chest_pid" 
    default="true"/>

    <!-- ======================== END OF SETTINGS ========================== -->

    <!-- Chest modules (dependency): -->
    <group if="$(arg launch_chest)">
        <include 
        file="$(find gopher_ros_clearcore)/launch/chest.launch">
            <arg 
            name="start_position" 
            value="$(arg start_position)"/>
        </include>
    </group>

    <!-- Chest PID control module (dependency): -->
    <group if="$(arg launch_chest_pid)">
        <include 
        file="$(find gopher_ros_clearcore)/launch/chest_pid.launch"/>
    </group>

    <!-- Proximity control module: -->
    <node 
    pkg="gopher_ros_clearcore" 
    type="proximity_control.py" 
    name="proximity_control" 
    output="screen">
        <param 
        name="node_frequency" 
        value="$(arg node_frequency)"/>
        <param 
        name="arm_center_of_workspace_z" 
        value="$(arg arm_center_of_workspace_z)"/>
        <param 
        name="activation_boundary_z" 
        value="$(arg activation_boundary_z)"/>
    </node>
</launch>
